doc/nrf/config_and_build/config_and_build_system.rst

```diff
@@ -9,28 +9,159 @@ Build and configuration system
 
-* :ref:`zephyr:application` is a complete guide to application development with Zephyr, including the build and configuration system.
-* :ref:`zephyr:cmake-details` describes in-depth the usage of CMake for Zephyr-based applications.
-* :ref:`zephyr:application-kconfig` contains a guide for Kconfig usage in applications.
-* :ref:`zephyr:set-devicetree-overlays` explains how to use devicetree and its overlays to customize an application's devicetree.
-* :ref:`board_porting_guide` is a complete guide to porting Zephyr to your own board.
+   * - Source
+     - File types
+     - Usage
+     - Available dedicated editing tools
+     - Additional information
+   * - :ref:`CMake <zephyr:build_overview>`
+     - :file:`CMakeLists.txt`, :file:`.cmake`
+     - Build system generator.
+     - n/a
+     - CMake is used to build your application together with the Zephyr kernel.
+   * - :ref:`Devicetree <zephyr:dt-guide>`
+     - :file:`.dts`, :file:`.dtsi`, :file:`.overlay`
+     - Hardware description language.
+     - `Devicetree Visual Editor <How to work with Devicetree Visual Editor_>`_
+     - Devicetree Visual Editor is part of the |nRFVSC|. You still need to be familiar with the devicetree language to use it.
+   * - :ref:`Kconfig <zephyr:application-kconfig>`
+     - :file:`Kconfig`, :file:`prj.conf`, :file:`.config`
+     - Software configuration system also used in the Linux kernel.
+     - `Kconfig GUI <Configuring with nRF Kconfig_>`_, :ref:`menuconfig and guiconfig <zephyr:menuconfig>`
+     - Kconfig GUI is part of the |nRFVSC|.
+   * - :ref:`partition_manager`
+     - :file:`pm.yml`, :file:`pm_static.yml`
+     - Memory layout configuration system.
+     - :ref:`partition_manager` script
+     - Partition Manager is an |NCS| configuration system that is not available in Zephyr.
+
+Each of these systems comes with a specialized syntax and purpose.
+See the following sections for more information.
+To read more about Zephyr's configuration system and its role in the application development, see :ref:`zephyr:build_overview` and :ref:`zephyr:application` in the Zephyr documentation.
+
+When you :ref:`create_application`, the configuration files for each of these systems are created in the :ref:`application directory <create_application_structure>`: :file:`CMakeLists.txt` for CMake, :file:`app.overlay` for Devicetree, :file:`prj.conf` for Kconfig, and :file:`partitions.yml` for Partition Manager (if enabled).
+You can then edit them according to your needs (see :ref:`modifying`).
+
+When you start building, a CMake build is executed in two stages: configuration phase and building phase.
+
+.. _configuration_system_overview_config:
+
+Configuration phase
+===================
+
+During this phase, CMake executes build scripts from :file:`CMakeLists.txt` and gathers configuration from different sources to generate the final build scripts at the application source directory and create a model of the build for the specified build target.
```

Comment:
> to generate the final build scripts at the application source directory

not sure exactly what you try to describe but sounds wrong to me.
A benefit of CMake is exactly that you avoid in-source builds and can place the build folder anywhere you like, for example: `cmake -B/foo/build sample/bar` will place the generated build files in `/foo/build` which is different from `sample/bar` which holds the source files.

Reply:
Rephrased to:
> During this phase, CMake executes build scripts from :file:`CMakeLists.txt` and gathers configuration from different sources to generate the final build scripts and create a model of the build for the specified build target.